base/iterators.jl

```diff
@@ -441,6 +442,31 @@ julia> collect(Iterators.rest([1,2,3,4], 2))
 """
 rest(itr,state) = Rest(itr,state)
 
+"""
+    peel(iter)
```

Comment:
Cute name, but I wonder if this shouldn't be called `headtail` or something more explicit like that. Is "peel" a traditional name for this kind of iterator?

Reply:
https://en.wikipedia.org/wiki/Loop_splitting#Loop_peeling standard among compiler people I guess?

Reply:
Maybe call it `henchmen`  then?

Reply:
Maybe, but that term has always meant because of type stability for me, while this is just a generic iteration peel.

Reply:
Was joking, of course. But I still think `headtail` might be better.

Reply:
I guess it's not exported though so it doesn't matter that much at this point.